StockTrader/src/main/java/com/restResource/StockTrader/service/BuyTriggerService.java

```diff
@@ -0,0 +1,117 @@
+package com.restResource.StockTrader.service;
+
+import com.restResource.StockTrader.entity.BuyTrigger;
+import com.restResource.StockTrader.entity.CommandType;
+import com.restResource.StockTrader.entity.Quote;
+import com.restResource.StockTrader.entity.logging.ErrorEventLog;
+import com.restResource.StockTrader.repository.AccountRepository;
+import com.restResource.StockTrader.repository.BuyTriggerRepository;
+import com.restResource.StockTrader.repository.InvestmentRepository;
+import org.springframework.core.task.TaskExecutor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+@Service
+public class BuyTriggerService {
+
+    private QuoteService quoteService;
+    private BuyTriggerRepository buyTriggerRepository;
+    private TaskExecutor taskExecutor;
+    private AccountRepository accountRepository;
+    private InvestmentRepository investmentRepository;
+    private LoggingService loggingService;
+
+    public BuyTriggerService(LoggingService loggingService,
+                             QuoteService quoteService,
+                             BuyTriggerRepository buyTriggerRepository,
+                             TaskExecutor taskExecutor,
+                             InvestmentRepository investmentRepository,
+                             AccountRepository accountRepository){
+        this.quoteService = quoteService;
+        this.buyTriggerRepository = buyTriggerRepository;
+        this.taskExecutor = taskExecutor;
+        this.accountRepository = accountRepository;
+        this.investmentRepository = investmentRepository;
+        this.loggingService = loggingService;
+    }
+
+    public void start(String userId, String stockSymbol, Integer stockCost, Integer transactionNum) {
+        taskExecutor.execute(new Runnable() {
+            @Override
+            public void run(){
+                //wait for both rest calls to complete, verify sufficient funds.
+                for (;;) {
+
+                    try {
+                        Thread.sleep(5000);
+                    } catch(InterruptedException e) {
+                        throw new IllegalArgumentException(
+                                "Error with Thread.sleep in "+Thread.currentThread().getName());
+                    }
+
+                    Optional<BuyTrigger> buyStockSnapshot = buyTriggerRepository.findByUserIdAndStockSymbol(userId, stockSymbol);
+
+                    if (!buyStockSnapshot.isPresent()) { //we dont have an entry yet, so continue waiting
```

Comment:
Since we require the user to have created a BUY_AMOUNT, this should throw an error, not just continue to loop